File: cplusplus/fashion.cpp
#include <iostream>
#include <vector>

using namespace std;


void sort (vector <int> &vec)
{
	for (int i = 0; i < vec.size(); ++i)
	{
		for (int j = 0; j < vec.size() - 1 - i; ++j)
		{
			if (vec[j] > vec[j+1])
			{
				int temp = vec[j];
				vec[j] = vec[j+1];
				vec[j+1] = temp;
			}
		}
	}
}


int main()
{
	int t; // t - number of test cases
	cin >> t;
	cin.get();
	for (int i = 0; i < t; ++i)
	{
		vector <int> male; // male - hotness levels of men
		vector <int> female; // female - hotness levels of female
		int N; // N - number of participants
		cin >> N;
		cin.get();
		for (int j = 0; j < N; ++j)
		{
			int temp; // temp - temporary for each participant
			cin >> temp;
			male.push_back (temp);
		}
		cin.get();
		for (int j = 0; j < N; ++j)
		{
			int temp;
			cin >> temp;
			female.push_back (temp);
		}
		sort (male);
		sort (female);
		int total = 0; //total - sum of hotness bonds
		for (int j = 0; j < N; ++j)
		{
			total += male[j] * female[j];
		}
		cout << total << endl;
	}
	return 0;
}

File: python/fashion.py
def merge (v, begin, end):
	middle = (begin + end) // 2
	i = 0
	j = middle + 1
	vs = []
	while i <= middle and j <= end:
		if v[i] < v[j]:
			vs.append (v[i])
			i += 1
		else:
			vs.append (v[j])
			j += 1
	while i <= middle:
		vs.append (v[i])
		i += 1
	while j <= end:
		vs.append (v[j])
		j += 1
	for i in range (begin, end + 1):
		v[i] = vs[i - begin]


def sort (v, begin, end):
	if (end > begin):
		middle = (begin + end) // 2
		sort (v, begin, middle)
		sort (v, middle + 1, end)
		merge (v, begin, end)


t = int (raw_input())
while t:
	N = int (raw_input())
	male = [int (x) for x in (raw_input()).split()]
	female = [int (x) for x in (raw_input()).split()]
	sort (male, 0, N-1)
	sort (female, 0, N-1)
	total = 0
	for i in range (0, N):
		total += male[i] * female[i]
	print total
	t -= 1
